osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/test/java/org/opensmartgridplatform/adapter/protocol/iec60870/testutils/matchers/GetLightSensorStatusResponseMessageMatcher.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.iec60870.testutils.matchers;
+
+import org.mockito.ArgumentMatcher;
+import org.opensmartgridplatform.dto.valueobjects.LightSensorStatusDto;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+import org.opensmartgridplatform.shared.infra.jms.ProtocolResponseMessage;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessageResultType;
+
+public class GetLightSensorStatusResponseMessageMatcher implements ArgumentMatcher<ProtocolResponseMessage> {
```

Comment:
I am a bit confused why some choices are made.
For instance, why is there an extra field for the LightSensorStatusDto, and why are some values in the matches method hard coded, while others compare the values from `argument` with those from `this.responseMessage`.
Couldn't all values be compared based on looking at values of `argument` and `this.responseMessage`?
Maybe this class could even be a more generic ProtocolResponseMessageMatcher then?
Am I missing something?

Reply:
I guess what happened, was that I started out with creating a matcher for LightSensorStatusDto, but later on noticed that it was contained within a protocol response message...
Anyway, I've refactored it into a ProtocolResponseMessageMatcher.